File: Readability Python/readability.py
from cs50 import get_string

# getting input from user
text = get_string("Text: ")
l = len(text)

# keeping track of words, letters, and sentences
words = 1
letters = 0
sentences = 0

# counting words, letter, and sentences per text
for i in range(l):
    if text[i].isalpha():
        letters += 1
    elif text[i] == " ":
        words += 1
    elif text[i] in [".", "?", "!"]:
        sentences += 1
    
# calculating Coleman-Liau index's L and S values
L = (letters / words) * 100
S = (sentences / words) * 100

# calculating Coleman-Liau index
grade = round(0.0588 * L - 0.296 * S - 15.8)

if grade < 1:
    print("Before Grade 1")
elif grade > 16:
    print("Grade 16+")
else:
    print(f"Grade {grade}")

File: Readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    // Setting an itial values for letters, words, sentences.
    int letters = 0;
    int words = 1;
    int sentences = 0;
    
    // Prompting user for text.
    string text = {get_string("Text: ")};
    
    // Getting length of the text.
    int length = strlen(text);
    
    // Looping over to get words, letters, and sentences.
    for (int i = 0; i < length; i++)
    {
        // Checking if current char is alphabet.
        if (isalpha(text[i]))
        {
            letters ++;
        }
        
        // Checking for spaces to detect end of words.
        if (text[i] == ' ')
        {
            words ++;
        }
        
        // Checking for end of sentences.
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences ++;
        }
    }
    
    float L = ((float) letters / words) * 100;
    float S = ((float) sentences / words) * 100;
    
    // Calculating Coleman-Liau index.
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    
    
    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }

}
